### PR TITLE
fix: harden `ASTVisitor` with default args

### PR DESCRIPTION
**Pull request by obycode**
Since the AST may be invalid, retrieving args from a list is not safe, since we cannot guarantee that the correct argument count is present. This commit adds a default value to fall back on so that the analysis may continue without crashing.

Fixes: #173

### CODE REVIEW

**Review by lgalabru — CHANGES_REQUESTED** [review]
I think we have one remaining undesired panic.
Not blocking, but I wonder if we should think about a design that would let us ensure that all these operations are correct by construction, with a macro or something? Maybe something to think about.
Thanks @obycode!

File: src/analysis/ast_visitor.rs [hunk 1]
```diff
@@ -137,118 +184,234 @@ pub trait ASTVisitor<'a> {
...
+                        ),
+                        AsMaxLen => {
+                            match args.get(1).unwrap_or(&DEFAULT_EXPR).match_literal_value() {
+                                Some(Value::UInt(length)) => self.traverse_as_max_len(
+                                    expr,
+                                    args.get(0).unwrap_or(&DEFAULT_EXPR),
+                                    *length,
+                                ),
+                                _ => panic!(
+                                    "check_checker: invalid length expression in as-max-len?"
```
[the remark this answers is not in the document]
Comment (obycode) [reply src/analysis/ast_visitor.rs]: 👍

**Comment by obycode** [comment]
> I think we have one remaining undesired panic.
Fixed.
> Not blocking, but I wonder if we should think about a design that would let us ensure that all these operations are correct by construction, with a macro or something? Maybe something to think about.
Agreed. I opened #177 for this.

Thanks!

**Review by lgalabru — APPROVED** [review]
Thanks @obycode!

**Review by lgalabru — APPROVED** [review]

**Pull request merged**